Python/ceval_macros.h

```diff
@@ -358,7 +358,9 @@ do {                                                   \
     frame = tstate->current_frame;                     \
     stack_pointer = _PyFrame_GetStackPointer(frame);   \
     if (next_instr == NULL) {                          \
-        next_instr = frame->instr_ptr;                 \
+        /* gh-140104: The exception handler expects frame->instr_ptr
+            to be pointing to next_instr, not this_instr! */ \
+        next_instr = frame->instr_ptr + 1 + _PyOpcode_Caches[_PyOpcode_Deopt[frame->instr_ptr->op.code]];                 \
```

Comment:
```suggestion
        next_instr = frame->instr_ptr + 1;
```
Just `+1` to compensate for the the -1 in `label(exception_unwind)`

Pointing to the next instruction does work, but it is unnecessary. As long as `next_inst -1` points into the current instruction, unwinding works correctly.